File: notification-service/src/main/java/com/eirsteir/coffeewithme/notification/service/NotificationService.java
package com.eirsteir.coffeewithme.notification.service;

import com.eirsteir.coffeewithme.notification.dto.NotificationDto;
import java.util.List;
import org.springframework.data.domain.Pageable;

public interface NotificationService {

  NotificationDto updateNotificationToRead(NotificationDto notificationDto);

  List<NotificationDto> findAllByUserId(Long id, Pageable pageable);
}

File: notification-service/src/main/kotlin/com/eirsteir/coffeewithme/notification/service/NotificationService.kt
package com.eirsteir.coffeewithme.notification.service

import com.eirsteir.coffeewithme.notification.dto.NotificationDto
import org.springframework.data.domain.Pageable

interface NotificationService {
    fun updateNotificationToRead(notificationDto: NotificationDto): NotificationDto
    fun findAllByUserId(id: Long, pageable: Pageable): MutableList<NotificationDto>
}